src/components/DataTable/index.jsx

```diff
@@ -62,48 +56,112 @@ class DataTable extends Component {
+        </TableSortLabel>
+      </TableCell>
+    );
+  };
+
+  componentDidMount = async () => {
+    await this.props.loadNextPage();
+  };
+
   render() {
     const {
       classes,
-      renderRow,
+      columns,
+      hasNextPage,
+      isNextPageLoading,
+      loadNextPage,
+      cellRenderer,
+      onHeaderClick,
       sortByHeader,
       sortDirection,
-      headers,
-      items,
+      rowCount,
+      ...tableProps
     } = this.props;
-    const colSpan = (headers && headers.length) || 0;
+    const itemCount = hasNextPage ? rowCount + 1 : rowCount;
+    const loadMoreItems = isNextPageLoading ? () => {} : loadNextPage;
+    const isItemLoaded = ({ index }) => !hasNextPage || index < rowCount;
 
     return (
-      <Table className={classes.table} aria-labelledby="tableTitle">
-        {headers && (
-          <TableHead>
-            <TableRow>
-              {headers.map(header => (
-                <TableCell key={`table-header-${header}`}>
-                  <TableSortLabel
-                    id={header}
-                    active={header === sortByHeader}
-                    direction={sortDirection || 'desc'}
-                    onClick={this.handleHeaderClick}>
-                    {header}
-                  </TableSortLabel>
-                </TableCell>
-              ))}
-            </TableRow>
-          </TableHead>
+      <InfiniteLoader
+        isRowLoaded={isItemLoaded}
+        loadMoreRows={loadMoreItems}
+        rowCount={itemCount}>
+        {({ onRowsRendered, ref }) => (
+          <AutoSizer>
+            {({ width }) => (
+              <WindowScroller>
+                {({ height, isScrolling, onChildScroll, scrollTop }) => (
+                  <Table
+                    autoHeight
+                    height={height}
+                    width={width}
+                    onRowsRendered={onRowsRendered}
+                    ref={ref}
+                    rowCount={rowCount}
+                    isScrolling={isScrolling}
+                    onScroll={onChildScroll}
+                    {...tableProps}
+                    rowClassName={this.getRowClassName}
+                    scrollTop={scrollTop}
+                    noRowsRenderer={() => (
+                      <TableCell
+                        style={{
+                          verticalAlign: 'middle',
+                          height: 56,
+                        }}
+                        width={columns.reduce(
+                          (sum, { width }) => sum + width,
+                          0
+                        )}>
+                        <em style={{ color: 'black' }}>
+                          No items for this page.
+                        </em>
+                      </TableCell>
+                    )}>
+                    {columns.map(({ dataKey, ...other }, index) => (
+                      <Column
+                        key={dataKey}
+                        headerRenderer={headerProps =>
```

Comment:
As the suggestion above to remove `this.headerRenderer` we can make it into something like below

```suggestion
                        headerRenderer={({ label }) => (
                          <TableCell
                            className={classes.flexContainer}
                            variant="head"
                            style={{ height: headerHeight }}>
                            <TableSortLabel
                              id={label}
                              active={label === sortByHeader}
                              direction={sortDirection || 'desc'}
                              onClick={this.handleHeaderClick}>
                              {label}
                            </TableSortLabel>
                          </TableCell>
                        )}
```